lib/ts/xmlhttprequest.ts

```diff
@@ -70,6 +66,21 @@ export function addInterceptorsToXMLHttpRequest() {
 
         const eventHandlers: Map<keyof XMLHttpRequestEventMap, Set<XHREventListener<any>>> = new Map();
 
+        let delayedQueue = firstEventLoopDone;
```

Comment:
can you explain this? Add comments here in code too.

Reply:
Added comments but basically we needed to do that to make XHR throw. Previously it was just logging the error to console.

Reply:
I dont see any comments.. not sure what you added. I have no idea what this change in XHR is supposed to even mean.

Reply:
The comment in on line 73, because that's what changed.

Line 69-70 has been moved down. They haven't changed.

Reply:
Previously, we were calling `console.error` if any error occurred in `delayIfNecessary` function but that would supress the error, instead we wanted it to throw. That's why I added the call to `onerror`.